lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GDNN.cpp

```diff
@@ -39,15 +39,29 @@ int DNNMsg::DecodeDNNMsg(
   dnn_message->len = ielen;
   MLOG(MDEBUG) << "len : " << static_cast<int>(dnn_message->len);
 
-  uint8_t dnn_len = 0;
+  uint8_t dnn_length = 0;
+  uint8_t dnn_len    = 0;
   DECODE_U8(buffer + decoded, dnn_len, decoded);
   MLOG(MDEBUG) << "dnn_len : " << static_cast<int>(dnn_len);
 
   memcpy(dnn_message->dnn, buffer + decoded, dnn_len);
-
+  dnn_length += dnn_len;
   decoded = decoded + dnn_len;
-  MLOG(MDEBUG) << "dnn str : " << dnn_message->dnn;
 
+  while (dnn_length + 1 < ielen) {
```

Comment:
why is this a while loop? how is the dot separated dnn is encoded as IE?

Reply:
Hi Ulas,
The dotted DNN is in the format label1.label2...label- n.
Its encoded in the format (total length, label1 length , label1, label2 length, label 2, ....label-n length, label-n)
<img width="396" alt="dottendnn" src="https://user-images.githubusercontent.com/94469973/148165623-14a36755-dee3-4785-b3b5-2e54a1500f1a.PNG">

Reply:
thanks for the clarification.